### ISSUE TITLE
HMM model 2 broken

### ISSUE THREAD

**Issue by jpchen** [uber/pyro#1523]
model 2 in the hmm example crashes with no error message when calling `train_loss = elbo.loss(model, guide, sequences, lengths, args, include_prior=False)` on cpu. the other models seem to work fine.

**Comment by fritzo** [comment]
Can you please add steps to reproduce? It surprises me since all models should be tested via tests/test_examples.py

**Comment by jpchen** [comment]
`python examples/hmm.py --model 2 [-n 1]`
@neerajprad verified that it works on pytorch master but breaks on release for both python 2 and 3.

**Comment by fritzo** [comment]
The reason the tests do not fail is that the tests run truncated versions `--truncate 10` and the error is not triggered when truncating. Reasons we run tests truncated include:
- truncated tests run faster
- until #1521 is merged, non-truncated tests will only pass on PyTorch 0.4 CPU (fail 0.4 GPU, 1.0 CPU, and 1.0 GPU)
